# Исследование надёжности заёмщиков



In [1]:
import pandas as pd

In [2]:
data = pd.read_csv('/datasets/data.csv')

In [3]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21525 entries, 0 to 21524
Data columns (total 12 columns):
children            21525 non-null int64
days_employed       19351 non-null float64
dob_years           21525 non-null int64
education           21525 non-null object
education_id        21525 non-null int64
family_status       21525 non-null object
family_status_id    21525 non-null int64
gender              21525 non-null object
income_type         21525 non-null object
debt                21525 non-null int64
total_income        19351 non-null float64
purpose             21525 non-null object
dtypes: float64(2), int64(5), object(5)
memory usage: 2.0+ MB


In [4]:
data.dtypes

children              int64
days_employed       float64
dob_years             int64
education            object
education_id          int64
family_status        object
family_status_id      int64
gender               object
income_type          object
debt                  int64
total_income        float64
purpose              object
dtype: object

In [5]:
print(data)

       children  days_employed  dob_years education  education_id  \
0             1   -8437.673028         42    высшее             0   
1             1   -4024.803754         36   среднее             1   
2             0   -5623.422610         33   Среднее             1   
3             3   -4124.747207         32   среднее             1   
4             0  340266.072047         53   среднее             1   
...         ...            ...        ...       ...           ...   
21520         1   -4529.316663         43   среднее             1   
21521         0  343937.404131         67   среднее             1   
21522         1   -2113.346888         38   среднее             1   
21523         3   -3112.481705         38   среднее             1   
21524         2   -1984.507589         40   среднее             1   

          family_status  family_status_id gender income_type  debt  \
0       женат / замужем                 0      F   сотрудник     0   
1       женат / замужем        

In [6]:
data.head(10)

children  days_employed  dob_years education  education_id  \
0         1   -8437.673028         42    высшее             0   
1         1   -4024.803754         36   среднее             1   
2         0   -5623.422610         33   Среднее             1   
3         3   -4124.747207         32   среднее             1   
4         0  340266.072047         53   среднее             1   
5         0    -926.185831         27    высшее             0   
6         0   -2879.202052         43    высшее             0   
7         0    -152.779569         50   СРЕДНЕЕ             1   
8         2   -6929.865299         35    ВЫСШЕЕ             0   
9         0   -2188.756445         41   среднее             1   

      family_status  family_status_id gender income_type  debt   total_income  \
0   женат / замужем                 0      F   сотрудник     0  253875.639453   
1   женат / замужем                 0      F   сотрудник     0  112080.014102   
2   женат / замужем                 0      M   сотрудник     0  145885.952297   
3   женат / замужем                 0      M   сотрудник     0  267628.550329   
4  гражданский брак                 1      F   пенсионер     0  158616.077870   
5  гражданский брак                 1      M   компаньон     0  255763.565419   
6   женат / замужем                 0      F   компаньон     0  240525.971920   
7   женат / замужем                 0      M   сотрудник     0  135823.934197   
8  гражданский брак                 1      F   сотрудник     0   95856.832424   
9   женат / замужем                 0      M   сотрудник     0  144425.938277   

                      purpose  
0               покупка жилья  
1     приобретение автомобиля  
2               покупка жилья  
3  дополнительное образование  
4             сыграть свадьбу  
5               покупка жилья  
6           операции с жильем  
7                 образование  
8       на проведение свадьбы  
9     покупка жилья для семьи

In [7]:
data.tail(10)

children  days_employed  dob_years       education  education_id  \
21515         1    -467.685130         28         среднее             1   
21516         0    -914.391429         42          высшее             0   
21517         0    -404.679034         42          высшее             0   
21518         0  373995.710838         59         СРЕДНЕЕ             1   
21519         1   -2351.431934         37  ученая степень             4   
21520         1   -4529.316663         43         среднее             1   
21521         0  343937.404131         67         среднее             1   
21522         1   -2113.346888         38         среднее             1   
21523         3   -3112.481705         38         среднее             1   
21524         2   -1984.507589         40         среднее             1   

          family_status  family_status_id gender income_type  debt  \
21515   женат / замужем                 0      F   сотрудник     1   
21516   женат / замужем                 0      F   компаньон     0   
21517  гражданский брак                 1      F   компаньон     0   
21518   женат / замужем                 0      F   пенсионер     0   
21519         в разводе                 3      M   сотрудник     0   
21520  гражданский брак                 1      F   компаньон     0   
21521   женат / замужем                 0      F   пенсионер     0   
21522  гражданский брак                 1      M   сотрудник     1   
21523   женат / замужем                 0      M   сотрудник     1   
21524   женат / замужем                 0      F   сотрудник     0   

        total_income                            purpose  
21515  109486.327999              заняться образованием  
21516  322807.776603               покупка своего жилья  
21517  178059.553491       на покупку своего автомобиля  
21518  153864.650328               сделка с автомобилем  
21519  115949.039788  покупка коммерческой недвижимости  
21520  224791.862382                  операции с жильем  
21521  155999.806512               сделка с автомобилем  
21522   89672.561153                       недвижимость  
21523  244093.050500       на покупку своего автомобиля  
21524   82047.418899              на покупку автомобиля

In [8]:
print(data['days_employed'].isnull().sum())

2174


In [9]:
print(data['total_income'].isnull().sum())

2174


In [10]:
print(data['total_income'].max())
print(data['total_income'].min())
print(data['total_income'].mean())
print(data['total_income'].median())

2265604.028722744
20667.26379327158
167422.30220817294
145017.93753253992


In [11]:
print(data['dob_years'].max())
print(data['dob_years'].min())

75
0


In [12]:
print(data['children'].max())
print(data['children'].min())

20
-1


In [13]:
print(data['days_employed'].max())
print(data['days_employed'].min())

401755.40047533
-18388.949900568383


**Вывод**

В файле data.csv есть таблица с 21525 записью с данными по клиентам. Количество столбцов 12 - описательные характеристики по каждому клиенту, из которых 8 столбцов категориальные (education, education_id, family_status, family_status_id, gender, income_type, debt, purpose) и 4 количественные (children, days_employed, dob_years, total_income).

Качество данных. В столбцах days_employed и total_income 2174 пропущенных значений, что составляет 10% от общей выборки, а это довольно много. Необходимо будет исследовать являются ли эти пропуски случайными. Судя по статистике в столбце total_income более менее реальные данные.

В столбце с возрастом dob_years есть нулевые значения, потребуется обработка этих данных. В столбце с количеством детей children есть отрицательные и чрезмерно большие значение, что также потребует обработки.

В столбце количество дней трудового стажа days_employed есть нереально большие и отрицательные значения. В столбце с образованием education данные записаны в разнобой - какие-то с верхним регистром, какие-то с заглавной буквы.

Также, тип столбца dob_years установлен как вещественный, необходимо будет его конвертировать в целый. 

## Шаг 2. Предобработка данных

### Обработка пропусков

Обработка пропусков в столбцах days_employed и total_income. Так как количество пропусков  в этом столбце совпадает с количеством пропусков в столбце с total_income, проверяем нет ли взаимосвязи между этими пропусками.

In [14]:
i = 0  # счетчик
for m in range(len(data)):
    if pd.isna(data.loc[m, 'days_employed']) == pd.isna(data.loc[i, 'total_income']) == True:
        i += 1
i

0

Гипотеза подтвердилась, пропущенные значения совпадают. Из теоретической части: "Пропуски в количественных переменных заполняют характерными значениями. Это значения, характеризующие состояние выборки, - набора данных, выбранных для проведения исследования. Чтобы примерно оценить типичные значения выборки, годятся среднее арифметическое или медиана." Заменим пропуски медианными значениями в возрасте, а для этого определим возрастные категории клиентов.

In [15]:
data['dob_years'].describe()

count    21525.000000
mean        43.293380
std         12.574584
min          0.000000
25%         33.000000
50%         42.000000
75%         53.000000
max         75.000000
Name: dob_years, dtype: float64

In [16]:
stat = data['dob_years'].describe()
stat[4:7]

25%    33.0
50%    42.0
75%    53.0
Name: dob_years, dtype: float64

In [17]:
m1 = data['dob_years'].median()
m1

42.0

Добавим новый столбец в таблицу с возрастной группой через создание функции.

In [18]:
def age_category(age):
    if age <= stat[4]:
        return 'до 34'
    if stat[4] < age and age <= stat[5]:
        return '34-43'
    if stat[5] < age and age <= stat[6]:
        return '43-53'
    else:
        return '53+'
    
data['age_category'] = data['dob_years'].apply(age_category)

In [19]:
data.head()

children  days_employed  dob_years education  education_id  \
0         1   -8437.673028         42    высшее             0   
1         1   -4024.803754         36   среднее             1   
2         0   -5623.422610         33   Среднее             1   
3         3   -4124.747207         32   среднее             1   
4         0  340266.072047         53   среднее             1   

      family_status  family_status_id gender income_type  debt   total_income  \
0   женат / замужем                 0      F   сотрудник     0  253875.639453   
1   женат / замужем                 0      F   сотрудник     0  112080.014102   
2   женат / замужем                 0      M   сотрудник     0  145885.952297   
3   женат / замужем                 0      M   сотрудник     0  267628.550329   
4  гражданский брак                 1      F   пенсионер     0  158616.077870   

                      purpose age_category  
0               покупка жилья        34-43  
1     приобретение автомобиля        34-43  
2               покупка жилья        до 34  
3  дополнительное образование        до 34  
4             сыграть свадьбу        43-53

In [20]:
data.tail()

children  days_employed  dob_years education  education_id  \
21520         1   -4529.316663         43   среднее             1   
21521         0  343937.404131         67   среднее             1   
21522         1   -2113.346888         38   среднее             1   
21523         3   -3112.481705         38   среднее             1   
21524         2   -1984.507589         40   среднее             1   

          family_status  family_status_id gender income_type  debt  \
21520  гражданский брак                 1      F   компаньон     0   
21521   женат / замужем                 0      F   пенсионер     0   
21522  гражданский брак                 1      M   сотрудник     1   
21523   женат / замужем                 0      M   сотрудник     1   
21524   женат / замужем                 0      F   сотрудник     0   

        total_income                       purpose age_category  
21520  224791.862382             операции с жильем        43-53  
21521  155999.806512          сделка с автомобилем          53+  
21522   89672.561153                  недвижимость        34-43  
21523  244093.050500  на покупку своего автомобиля        34-43  
21524   82047.418899         на покупку автомобиля        34-43

In [21]:
mediana_days_empl = data.groupby('age_category')['days_employed'].median()
mediana_days_empl

age_category
34-43     -1766.842512
43-53     -1839.038911
53+      342859.373803
до 34     -1130.966487
Name: days_employed, dtype: float64

In [22]:
data.loc[(data['days_employed'].isnull()) & (data['age_category'] == '34-43'), 'days_employed'] = mediana_days_empl[0]
data.loc[(data['days_employed'].isnull()) & (data['age_category'] == '43-53'), 'days_employed'] = mediana_days_empl[1]
data.loc[(data['days_employed'].isnull()) & (data['age_category'] == '53+'), 'days_employed'] = mediana_days_empl[2]
data.loc[(data['days_employed'].isnull()) & (data['age_category'] == 'до 34'), 'days_employed'] = mediana_days_empl[3]

In [23]:
print(data['days_employed'].isnull().sum())

0


Заменим пропуски в столбце total_income медианными значениями для каждой возрастной группы

In [24]:
data.loc[data['total_income'].isnull(), 'age_category'].value_counts()

53+      565
43-53    545
34-43    544
до 34    520
Name: age_category, dtype: int64

In [25]:
mediana_income = data.groupby('age_category')['total_income'].median()
mediana_income

age_category
34-43    155562.307690
43-53    150839.438598
53+      130276.801649
до 34    145919.113211
Name: total_income, dtype: float64

In [26]:
data.loc[(data['total_income'].isnull()) & (data['age_category'] == '34-43'), 'total_income'] = mediana_income[0]
data.loc[(data['total_income'].isnull()) & (data['age_category'] == '43-53'), 'total_income'] = mediana_income[1]
data.loc[(data['total_income'].isnull()) & (data['age_category'] == '53+'), 'total_income'] = mediana_income[2]
data.loc[(data['total_income'].isnull()) & (data['age_category'] == 'до 34'), 'total_income'] = mediana_income[3]

In [27]:
print(data['total_income'].isnull().sum())

0


Решим вопрос с нулями в столбце dob_years.

In [28]:
data.loc[data['dob_years'] == 0, 'income_type'].value_counts()

сотрудник      55
пенсионер      20
компаньон      20
госслужащий     6
Name: income_type, dtype: int64

In [29]:
mediana_age = data.groupby('income_type')['dob_years'].median()
mediana_age

income_type
безработный        38.0
в декрете          39.0
госслужащий        40.0
компаньон          39.0
пенсионер          60.0
предприниматель    42.5
сотрудник          39.0
студент            22.0
Name: dob_years, dtype: float64

In [30]:
data.loc[(data['dob_years'] == 0) & (data['income_type'] == 'сотрудник'), 'dob_years'] = mediana_age[6]
data.loc[(data['dob_years'] == 0) & (data['income_type'] == 'компаньон'), 'dob_years'] = mediana_age[3]
data.loc[(data['dob_years'] == 0) & (data['income_type'] == 'пенсионер'), 'dob_years'] = mediana_age[4]
data.loc[(data['dob_years'] == 0) & (data['income_type'] == 'госслужащий'), 'dob_years'] = mediana_age[2]

In [31]:
print(data['dob_years'].max())
print(data['dob_years'].min())

75.0
19.0


Решим вопрос с количеством детей, заменим 20 на 2, и -1 на 1 (из соображений ошибки - человеческий фактор)

In [32]:
data.loc[data['children'] == 20, 'children'] = 2
data.loc[data['children'] == -1, 'children'] = 1

In [33]:
print(data['children'].max())
print(data['children'].min())

5
0


После всех изменений

In [34]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21525 entries, 0 to 21524
Data columns (total 13 columns):
children            21525 non-null int64
days_employed       21525 non-null float64
dob_years           21525 non-null float64
education           21525 non-null object
education_id        21525 non-null int64
family_status       21525 non-null object
family_status_id    21525 non-null int64
gender              21525 non-null object
income_type         21525 non-null object
debt                21525 non-null int64
total_income        21525 non-null float64
purpose             21525 non-null object
age_category        21525 non-null object
dtypes: float64(3), int64(4), object(6)
memory usage: 2.1+ MB


**Вывод**

Все строки стали одинкового количества = 21525. Тип данных стал корректным, значения в столбце возраст приведен к реальным 
данным, пропуски обработаны.

### Замена типа данных

In [36]:
data.dtypes

children              int64
days_employed       float64
dob_years           float64
education            object
education_id          int64
family_status        object
family_status_id      int64
gender               object
income_type          object
debt                  int64
total_income        float64
purpose              object
age_category         object
dtype: object

Тип столбца возраст клиента float64 поменяем на целый int64

In [37]:
data['dob_years'] = data['dob_years'].astype('int')

In [38]:
data.dtypes

children              int64
days_employed       float64
dob_years             int64
education            object
education_id          int64
family_status        object
family_status_id      int64
gender               object
income_type          object
debt                  int64
total_income        float64
purpose              object
age_category         object
dtype: object

Решим вопрос с разными значениями в столбце education

In [39]:
data['education'].value_counts()

среднее                13750
высшее                  4718
СРЕДНЕЕ                  772
Среднее                  711
неоконченное высшее      668
ВЫСШЕЕ                   274
Высшее                   268
начальное                250
Неоконченное высшее       47
НЕОКОНЧЕННОЕ ВЫСШЕЕ       29
НАЧАЛЬНОЕ                 17
Начальное                 15
ученая степень             4
УЧЕНАЯ СТЕПЕНЬ             1
Ученая степень             1
Name: education, dtype: int64

In [40]:
data['education'] = data['education'].str.lower()

In [41]:
data['education'].value_counts()

среднее                15233
высшее                  5260
неоконченное высшее      744
начальное                282
ученая степень             6
Name: education, dtype: int64

**Вывод**

Тип данных заменен на int64 (методом astype), исправилен регистр на корректный в столбце education. 

### Обработка дубликатов

In [42]:
count_duplicate = data.duplicated().sum()
count_duplicate

71

In [43]:
percent_duplicate = (data.duplicated().sum() / len(data))*100
percent_duplicate

0.32984901277584205

In [44]:
data = data.drop_duplicates().reset_index(drop = True)

In [45]:
print('Количество дубликатов:', data.duplicated().sum())

Количество дубликатов: 0


**Вывод**

Был найден 71 дубликат в таблице, дубликаты удалены методом drop_duplicates (используется при поиске идентичных строк).

### Лемматизация

In [46]:
from pymystem3 import Mystem
from collections import Counter

m = Mystem()

In [47]:
data['purpose'].value_counts()  # сделаем подсчет всех возможных целей кредита

свадьба                                   791
на проведение свадьбы                     768
сыграть свадьбу                           765
операции с недвижимостью                  675
покупка коммерческой недвижимости         661
операции с жильем                         652
покупка жилья для сдачи                   651
операции с коммерческой недвижимостью     650
жилье                                     646
покупка жилья                             646
покупка жилья для семьи                   638
строительство собственной недвижимости    635
недвижимость                              633
операции со своей недвижимостью           627
строительство жилой недвижимости          624
покупка недвижимости                      621
покупка своего жилья                      620
строительство недвижимости                619
ремонт жилью                              607
покупка жилой недвижимости                606
на покупку своего автомобиля              505
заняться высшим образованием      

После анализа всех целей кредита, видим, что все сводится к 7 категориям

In [48]:
categories = ["свадьба", "сдача", "строительство", "жилье", "образование", "автомобиль", "недвижимость"]

In [49]:
def lemmatize(text):
    lemmas = m.lemmatize(text)
    for i in categories:
        if i in lemmas:
            lemmas = i
    return lemmas
 
data['category_group'] = data['purpose'].apply(lemmatize)

In [50]:
data.head(5)

children  days_employed  dob_years education  education_id  \
0         1   -8437.673028         42    высшее             0   
1         1   -4024.803754         36   среднее             1   
2         0   -5623.422610         33   среднее             1   
3         3   -4124.747207         32   среднее             1   
4         0  340266.072047         53   среднее             1   

      family_status  family_status_id gender income_type  debt   total_income  \
0   женат / замужем                 0      F   сотрудник     0  253875.639453   
1   женат / замужем                 0      F   сотрудник     0  112080.014102   
2   женат / замужем                 0      M   сотрудник     0  145885.952297   
3   женат / замужем                 0      M   сотрудник     0  267628.550329   
4  гражданский брак                 1      F   пенсионер     0  158616.077870   

                      purpose age_category category_group  
0               покупка жилья        34-43          жилье  
1     приобретение автомобиля        34-43     автомобиль  
2               покупка жилья        до 34          жилье  
3  дополнительное образование        до 34    образование  
4             сыграть свадьбу        43-53        свадьба

In [51]:
data['category_group'].value_counts()   # количество по каждой категории кредита

недвижимость     4473
автомобиль       4306
образование      4013
жилье            3809
свадьба          2324
строительство    1878
сдача             651
Name: category_group, dtype: int64

Объединим категории сдача и строительство в инвестиционную цель, а категорию жилье включим в недвижимость.

In [52]:
data.loc[data['category_group'] == 'жилье', 'category_group'] = 'недвижимость'
data.loc[(data['category_group'] == 'строительство') | (data['category_group'] == 'сдача'), 'category_group'] = 'инвестиционная цель'

In [53]:
data['category_group'].value_counts()

недвижимость           8282
автомобиль             4306
образование            4013
инвестиционная цель    2529
свадьба                2324
Name: category_group, dtype: int64

**Вывод**

Было выделено несколько ключевых слов, которые характеризовали цели кредита, на основе этих ключевых слов созданы категории. Далее создана еще более крупная категория.  

### Категоризация данных

Проведем категоризацию по доходам

In [54]:
stat = data['total_income'].describe()
stat[4:7]

25%    107623.856558
50%    145919.113211
75%    195813.983455
Name: total_income, dtype: float64

Создадим функцию для определения категории доходов

In [55]:
def _income_category(income):
    if income <= stat[4]:
        return 1
    if income > stat[4] and income <= stat[5]:
        return 2
    if income > stat[5] and income <= stat[6]:
        return 3
    else:
        return 4
    
data['income_category'] = data['total_income'].apply(_income_category)

In [56]:
data.head(5)

children  days_employed  dob_years education  education_id  \
0         1   -8437.673028         42    высшее             0   
1         1   -4024.803754         36   среднее             1   
2         0   -5623.422610         33   среднее             1   
3         3   -4124.747207         32   среднее             1   
4         0  340266.072047         53   среднее             1   

      family_status  family_status_id gender income_type  debt   total_income  \
0   женат / замужем                 0      F   сотрудник     0  253875.639453   
1   женат / замужем                 0      F   сотрудник     0  112080.014102   
2   женат / замужем                 0      M   сотрудник     0  145885.952297   
3   женат / замужем                 0      M   сотрудник     0  267628.550329   
4  гражданский брак                 1      F   пенсионер     0  158616.077870   

                      purpose age_category category_group  income_category  
0               покупка жилья        34-43   недвижимость                4  
1     приобретение автомобиля        34-43     автомобиль                2  
2               покупка жилья        до 34   недвижимость                2  
3  дополнительное образование        до 34    образование                4  
4             сыграть свадьбу        43-53        свадьба                3

Категоризация по доходу выглядит

In [57]:
data['income_category'].value_counts()

2    5460
4    5364
1    5364
3    5266
Name: income_category, dtype: int64

Категоризация по целям - category_group

In [58]:
data['category_group'].value_counts()

недвижимость           8282
автомобиль             4306
образование            4013
инвестиционная цель    2529
свадьба                2324
Name: category_group, dtype: int64

Категоризация по возрасту - age_category

In [59]:
data['age_category'].value_counts()

до 34    5467
43-53    5448
34-43    5284
53+      5255
Name: age_category, dtype: int64

**Вывод**

Самая большая категория в возрасте до 34 лет. Самая распространенная цель кредита - недвижимость.

## Шаг 3. Ответьте на вопросы

- Есть ли зависимость между наличием детей и возвратом кредита в срок?

Категоризируем клиентов по наличию и отсутствию детей. Для этого создадим функцию.

In [60]:
def _has_children(children):
    if children > 0:
        return 1
    else:
        return 0
    
data['has_children'] = data['children'].apply(_has_children)

In [61]:
new_data = data.pivot_table(index = ['has_children'], values = 'debt')
new_data.head()

debt
has_children          
0             0.075438
1             0.092082

**Вывод**

При наличии детей вероятность задолженности растет. Однако, стоит учитывать, что в этой категории есть у кого и 3 и 4 ребенка, что может говорить о том, что таким клиентам проще увеличить просрочку по кредиту, нежели не купить детям необходимое.

- Есть ли зависимость между семейным положением и возвратом кредита в срок?

In [62]:
new_data = data.pivot_table(index = ['family_status'], columns = 'debt', values = 'gender', aggfunc = 'count')
new_data['coeff'] = new_data[1] / (new_data[0] + new_data[1])
new_data.sort_values('coeff', ascending = False)

debt                       0    1     coeff
family_status                              
Не женат / не замужем   2536  274  0.097509
гражданский брак        3763  388  0.093471
женат / замужем        11408  931  0.075452
в разводе               1110   85  0.071130
вдовец / вдова           896   63  0.065693

**Вывод**

Из значений вероятности (столбец coeff) видим, что если клиент был в браке, вероятность задолженности ниже. Брак заставляет отдавать долги :)

- Есть ли зависимость между уровнем дохода и возвратом кредита в срок?

In [63]:
new_data = data.pivot_table(index = ['income_category'], columns = 'debt', values = 'gender', aggfunc = 'count')
new_data['coeff'] = new_data[1] / (new_data[0] + new_data[1])
new_data.sort_values('coeff', ascending = False)

debt                0    1     coeff
income_category                     
3                4807  459  0.087163
2                4988  472  0.086447
1                4937  427  0.079605
4                4981  383  0.071402

**Вывод**

Из значений вероятности (столбец coeff) видим, что вероятность стать должником ниже у клиентов, зарабатывающих больше (что вполне логично).

- Как разные цели кредита влияют на его возврат в срок?

In [64]:
new_data = data.pivot_table(index = ['category_group'], columns = 'debt', values = 'gender', aggfunc = 'count')
new_data['coeff'] = new_data[1] / (new_data[0] + new_data[1])
new_data.sort_values('coeff', ascending = False)

debt                    0    1     coeff
category_group                          
автомобиль           3903  403  0.093590
образование          3643  370  0.092200
свадьба              2138  186  0.080034
инвестиционная цель  2333  196  0.077501
недвижимость         7696  586  0.070756

**Вывод**

Самая низкая вероятность задолженности по клиентам, которые приобретают недвижимость, в то время как самая высокая - у автокредитов.

## Шаг 4. Общий вывод

С точки зрения банка, самый надежный клиент - женатый, без детей, много зарабатывающий клиент; при идеальных условиях.